neo/Consensus/ConsensusService.cs

```diff
@@ -182,25 +182,27 @@ private void OnChangeViewReceived(ConsensusPayload payload, ChangeView message)
             // ChangeView messages include a Timestamp when the change view is sent, thus if a node restarts
             // and issues a change view for the same view, it will have a different hash and will correctly respond
             // again; however replay attacks of the ChangeView message from arbitrary nodes will not trigger an
-            // additonal recovery message response.
+            // additional recovery message response.
             if (!knownHashes.Add(payload.Hash)) return;
             if (message.NewViewNumber <= context.ViewNumber)
             {
-                bool shouldSendRecovery = false;
-                // Limit recovery to sending from `f` nodes when the request is from a lower view number.
-                int allowedRecoveryNodeCount = context.F();
-                for (int i = 0; i < allowedRecoveryNodeCount; i++)
+                if (!context.CommitSent())
```

Comment:
If Commit was sent, i send the recovery message right?

Reply:
It forces all nodes that are committed to always send recovery.

We were afraid that in the edge cases nodes would not be recovered because of indexed not matching (just `f` nodes would reply);

Reply:
If commit was sent (and the view requested is greater than or equal to the view in which committed), then yes, the node sends recovery. If there are commits in two different views, we want the highest view to definitely reply to minimize the chance of a split. 

I've been thinking about how to avoid the split situation. One mechanism is on start, when requesting recovery by sending ChangeView, the code could wait for a certain time to get the responses. This would allow it to receive the recovery messages potentially from both views, and then it can choose the higher one.

Also, I thought of one other solution that can work to automatically resolve the split:
if a node is committed in a view, and it sees a recovery with >= f+1 commits in a higher view, and it has seen <= f commits in its current view; that could be acceptable criteria for it to go ahead and accept recovery to a higher view and commit in the higher view.

However, I think how it is now in this PR it should likely hardly every stall or require manual intervention on behalf of CN operators.